pytorch_pfn_extras/_dynamo/_splitter.py

```diff
@@ -131,3 +149,49 @@ def _no_partition(
         bwd_graph.output(out_nodes)
         bwd_module = torch.fx.GraphModule(joint_module, bwd_graph)
         return (fwd_module, bwd_module)
+
+
+class ForwardOnly(_Splitter):
+    def __init__(
+        self,
+        parameter_names: Optional[List[str]] = None,
+    ):
+        if parameter_names is not None:
+            self._parameter_names = [
+                n.replace(".", "__dot__") for n in parameter_names
+            ]
+        else:
+            self._parameter_names = []
+
+    def partition(
+        self,
+        joint_module: torch.fx.GraphModule,
+        _joint_inputs: Any,
+        *,
+        num_fwd_outputs: int,
+    ) -> Tuple[torch.fx.GraphModule, torch.fx.GraphModule]:
+        fwd_graph, _ = default_partition(
+            joint_module,
+            _joint_inputs,
+            num_fwd_outputs=num_fwd_outputs,
+        )
+        # Change the input names in the fwd graph
+        primal_inputs: List[torch.fx.Node] = list(
+            filter(_is_primal, fwd_graph.graph.nodes)
+        )
+        for i, node in enumerate(primal_inputs):
+            node.name = (
+                self._parameter_names[i]
+                if i < len(self._parameter_names)
+                else f"input_{i - len(self._parameter_names)}"
+            )
+
+        # Remove the outputs
```

Comment:
I want some detailed comments about the 'output_node' that is being deleted here. I believe it's an intermediate variable for gradient calculation, but is it sufficient to just delete one?

Reply:
added comments explaining it!